src/android/com/mady/wifi/api/WifiHotSpots.java

```diff
@@ -251,30 +251,43 @@ public boolean connectToHotspot(String netSSID, String netPass, Integer authenti
     }
 
     /**
-     * Determines if current connection is a captive portal by checking if we get a 204 response
-     * from google. If we do not, we can assume this is do to a captive portal instead returning a 200 response.
-     * Logic taken from https://stackoverflow.com/questions/13958614/how-to-check-for-unrestricted-internet-access-captive-portal-detection
+     * This function works under the assumption that we have already determined that we are connected to a valid network.
+     * Determines if current connection is a captive portal by checking if we can access Apple's captive portal detection
+     * page. If we receive 'Success' in the request's response body, there is not
```

Comment:
```suggestion
     * page. If we receive 'Success' in the request's response body, there is no captive portal.
```

The wording is confusing. It seems to be missing something.